dispatch/apps/api/views.py

```diff
@@ -638,24 +638,35 @@ def list(self, request):
 #     except:
 #         return Response(request.data, status=status.HTTP_400_BAD_REQUEST)
 
-
-@api_view(['POST'])
 @permission_classes((AllowAny,))
-def user_authenticate(request):
+class AuthenticationViewSet(viewsets.GenericViewSet):
 
-    email = request.data.get('email', None)
-    password = request.data.get('password', None)
+    def user_authenticate(self, request):
+        email = request.data.get('email', None)
+        password = request.data.get('password', None)
 
-    user = authenticate(username=email, password=password)
+        user = authenticate(username=email, password=password)
 
-    if user is not None and user.is_active:
+        if user is not None and user.is_active:
 
-        (token, created) = Token.objects.get_or_create(user=user)
+            (token, created) = Token.objects.get_or_create(user=user)
 
-        data = {
-            'token': unicode(token)
-        }
+            data = {
+                'token': unicode(token)
+            }
 
-        return Response(data, status=status.HTTP_202_ACCEPTED)
-    else:
-        return Response({}, status=status.HTTP_400_BAD_REQUEST)
+            return Response(data, status=status.HTTP_202_ACCEPTED)
+        else:
+            return Response({}, status=status.HTTP_400_BAD_REQUEST)
+
+    def user_unauthenticate(self, request):
+
+        user_token = request.data
+        if user_token is None:
+            return Response({}, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            try:
+                Token.objects.get(user_token).delete()
+            except:
+                pass
+            return Response({}, status=status.HTTP_200_OK)
```

Comment:
[200 is the default status](http://www.django-rest-framework.org/api-guide/responses/#response) if none is provided; I think it's more clear to omit it, so that the cases where you specify a different status are more obvious, but hey that's up to you.